sale_product_set_sale_by_packaging/models/product_template.py

```diff
@@ -0,0 +1,51 @@
+# Copyright 2021 Camptocamp SA
+# @author Simone Orsi <simone.orsi@camptocamp.com>
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl)
+from odoo import api, fields, models
+
+
+class ProductTemplate(models.Model):
+    _inherit = "product.template"
+
+    # Just for UI purpose
+    sell_only_by_packaging_prod_set_tocheck = fields.Boolean(
+        help="If this flag is ON, "
+        "it means you have to check all product.set associated with it",
+        compute="_compute_sell_only_by_packaging_prod_set_tocheck",
+    )
+
+    @api.depends("sell_only_by_packaging")
+    def _compute_sell_only_by_packaging_prod_set_tocheck(self):
+        ids_to_check = self._product_ids_to_check()
+        for rec in self:
+            value = False
+            if rec.sell_only_by_packaging and ids_to_check:
+                value = set(rec.product_variant_ids.ids).intersection(ids_to_check)
+            rec.sell_only_by_packaging_prod_set_tocheck = value
+
+    def _product_ids_to_check(self):
+        """Retrieves product.product IDS to check on product sets.
+        """
+        ids_to_check = self.filtered("sell_only_by_packaging").product_variant_ids.ids
+        if not ids_to_check:
+            return []
+        query = """
```

Comment:
Is a search() with that limited domain (ids and product_packaging_id) has really poor performances to do it through query ?

Reply:
security is checked already by accessing the variants, IMO there's no need to rely on ORM if we can get this faster.
I could go for a search_read but it does not allow to pass `load="_classic_write"` to fetch only prod ids.

Reply:
About security, maybe a comment is welcome to explain this?

Reply:
I've remove the SQL to keep it clean.
Can you update your review?